bin/server.js

```diff
@@ -41,35 +42,49 @@ async function startServer () {
   app.use(cors({ origin: '*' }))
 
   // MIDDLEWARE END
-
   // Retrieve hash from BCH network and retrieve data from IPFS.
 
   // Get the latest hash off the BCH network.
   const hash = await bch.findHash()
+  // To-Do: Add some retry code here. If it doesn't find the hash on the first try,
+  // it should retry a few times. The npm p-retry library would be good to use here.
 
   // Exit if no hash is found.
   if (!hash) {
-    console.log(`Could not find IPFS hash associated with BCH address ${config.BCHADDR}`)
-    console.log(`Publish an IPFS hash using the memo-push tool before running this server.`)
-    // console.log(`Exiting`)
-    // process.exit()
+    console.log(
+      `Could not find IPFS hash associated with BCH address ${config.BCHADDR}`
+    )
+    console.log(
+      `Publish an IPFS hash using the memo-push tool before running this server.`
+    )
+    console.log(`Exiting`)
+    process.exit()
   }
 
-  console.log(`Retrieving and serving this IPFS hash: ${hash}`)
+  // Start IPFS
+  const ipfsNode = await ipfs.startIPFS()
+
+  // For ipfs Ready
+  ipfsNode.on('ready', async () => {
+    console.log(`Retrieving and serving this IPFS hash: ${hash}`)
 
-  // Get the content from the IPFS network.
-  shell.cd(`ipfs-data`)
-  shell.exec(`ipfs get ${hash}`)
-  shell.exec(`ipfs pin add ${hash}`)
+    // Bootstrap connection (for development purposes)
+    // Replace the multiaddr below with the IPFS node running on your local machine.
+    await ipfsNode.swarm.connect('/ip4/127.0.0.1/tcp/4001/ipfs/QmSgDzV1GeTg1tx4wU5WKjxMvT692xvt8FS14JdoDEgFjj')
```

Comment:
Feel free to remove this code or edit it as needed. While debugging the code, I ran a separate IPFS node using go-ipfs, with the command `ipfs daemon`. This line adds that node to the swarm, so that this instance of js-ipfs can bootstrap and download the IPFS data off my main computer.